pkg/daemon/daemon.go

```diff
@@ -1383,39 +1365,45 @@ func (dn *Daemon) Run(stopCh <-chan struct{}, exitCh <-chan error, errCh chan er
 			// This channel gets errors from auxiliary goroutines like loginmonitor and kubehealth
 			klog.Warningf("Got an error from auxiliary tools: %v", err)
 		case err := <-errCh:
-			klog.Errorf("Fatal error from auxiliary tools: %v", err)
+			klog.Errorf("Got an error from auxiliary tools: %v", err)
 			// we do not want to fail on any .HandleError call. Need to only fail when it is a watcher
 			// we might want to remove this last one. We will see.
-			if strings.Contains(strings.ToLower(err.Error()), "failed to watch") || strings.Contains(strings.ToLower(err.Error()), "unknown authority") || strings.Contains(strings.ToLower(err.Error()), "error on the server") {
-				if dn.deferKubeletRestart {
-					dn.deferKubeletRestart = false
-					err = os.Remove("/var/lib/kubelet/kubeconfig")
-					if err != nil {
-						return fmt.Errorf("could not remove kubelet's kubeconfig file: %v", err)
-					}
-					if err := runCmdSync("systemctl", "restart", "kubelet"); err != nil {
-						return err
-					}
-					if err := wait.PollUntilContextTimeout(context.TODO(), 10*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
-						_, err := os.ReadFile("/var/lib/kubelet/kubeconfig")
-						if err != nil && os.IsNotExist(err) {
-							klog.Warningf("Failed to get kubeconfig file: %v", err)
-							return false, nil
-						} else if err != nil {
-							return false, fmt.Errorf("unexpected error reading kubeconfig file, %v", err)
-						}
-
-						return true, nil
-					}); err != nil {
-						return fmt.Errorf("something went wrong while waiting for kubeconfig file to generate: %v", err)
-					}
+			if dn.deferKubeletRestart && (strings.Contains(strings.ToLower(err.Error()), "failed to watch") || strings.Contains(strings.ToLower(err.Error()), "unknown authority") || strings.Contains(strings.ToLower(err.Error()), "error on the server")) {
```

Comment:
For a second I thought this was doing the same thing, but the key difference is that `ErrAuxiliary` gets wrapped into a kubelet restart. I think this should be safe so let's give it a try